src/jit/morph.cpp

```diff
@@ -11416,6 +11416,20 @@ GenTreePtr Compiler::fgMorphSmpOp(GenTreePtr tree, MorphAddrContext* mac)
                 }
             }
 
+            // If gtOp1 is a GT_FIELD, we need to pass down the mac if
+            // its parent is GT_ADDR, since the address of the field
+            // is part of an ongoing address computation. Otherwise
+            // op1 represents the value of the field and so any address
+            // calculations it does are in a new context.
+            if ((op1->gtOper == GT_FIELD) && (tree->gtOper != GT_ADDR))
```

Comment:
The null checks that are being added are for the case where we add some offset to a base pointer (typically a GC pointer) and the amount that we add might push us past the set of unmapped pages that start at address 0.  So if for instance the base pointer was a nullptr and we added 10,000 to pushing up into a valid mapped page we would access some other valid memory and no generate an access violation.  It is also be a security issue if we allow this pattern to occur.